slug.go

```diff
@@ -19,6 +19,23 @@ type Meta struct {
 	Size int64
 }
 
+// IllegalSlugError indicates the provided slug (io.Writer for Pack, io.Reader
+// for Unpack) validates a rule about its contents. For example, an absolute or
+// external symlink.
+type IllegalSlugError struct {
+	Err error
+}
+
+// Error implements the error interface by returning an error string about the
+// issue with the underlying error.
```

Comment:
I would usually not document the implementation of a known interface like this. It's a hard sentence to read and I think stating that `IllegalSlugError` implements the `error` interface on the struct is sufficient.